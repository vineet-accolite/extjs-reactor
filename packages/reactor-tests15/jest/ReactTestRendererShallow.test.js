import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { reactify } from '@extjs/reactor';
const [ Container, Button, Grid, Column, Toolbar, Panel ] = reactify('Container', 'Button', 'Grid', 'Column', 'Toolbar', 'Panel');
 
describe('ReactTestRenderer shallow snapshots', () => {
    let renderer;

    beforeEach(() => {
        renderer = new ReactShallowRenderer()
    });
    
    it('should support shallow rendering', () => {
        const TestComponent = () => (
            <Container>
                <Panel title="Child"/>
                <ChildComponent/>
            </Container>
        );

        const ChildComponent = () => (
            <Container>
                <Button text="Button"/>
            </Container>
        )

        renderer.render(<TestComponent/>);
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });

    it('should support child html elements', () => {
        const TestComponent = () => (
            <Container>
                <div>Foo</div>
                <ChildComponent/>
            </Container>
        );

        const ChildComponent = () => (
            <Container>
                <Button text="Button"/>
            </Container>
        )

        renderer.render(<TestComponent/>);
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });
});