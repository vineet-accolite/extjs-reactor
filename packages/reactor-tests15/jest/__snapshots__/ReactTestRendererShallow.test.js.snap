// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactTestRenderer shallow snapshots should support child html elements 1`] = `
<Container>
  <div>
    Foo
  </div>
  <ChildComponent />
</Container>
`;

exports[`ReactTestRenderer shallow snapshots should support shallow rendering 1`] = `
<Container>
  <Panel
    title="Child"
  />
  <ChildComponent />
</Container>
`;
